lib/auth/machineid/machineidv1/bot_service.go

```diff
@@ -727,13 +727,20 @@ func botFromUserAndRole(user types.User, role types.Role) (*pb.Bot, error) {
 
 	expiry := botExpiryFromUser(user)
 
+	// We need to return a revision because the Terraform provider uses it to
+	// determine when the change has been applied to the cache. We do not use
+	// the revision for conditional updates, though, so the concatenation of
+	// the user and role's revisions is sufficient.
+	revision := user.GetMetadata().Revision + role.GetMetadata().Revision
```

Comment:
I wonder if we should concatenate these with a seperation character that would allow us to easily recover the revisions if we were ever to support conditional updates.

Reply:
Yeah, that's a fair shout! I wasn't sure if we assume a certain format for these, but I see it's backend-dependent.